ecs-agent/daemonimages/csidriver/driver/node_test.go

```diff
@@ -0,0 +1,161 @@
+	"context"
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/golang/mock/gomock"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	volumeID = "voltest"
+	nvmeName = "/dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_voltest"
+)
+
+func TestNodeGetVolumeStats(t *testing.T) {
+	testCases := []struct {
+		name     string
+		testFunc func(t *testing.T)
+	}{
+		{
+			name: "success normal",
+			testFunc: func(t *testing.T) {
+				mockCtl := gomock.NewController(t)
+				defer mockCtl.Finish()
+
+				mockMounter := NewMockMounter(mockCtl)
+				VolumePath := "./test"
+				err := os.MkdirAll(VolumePath, 0644)
+				if err != nil {
+					t.Fatalf("fail to create dir: %v", err)
+				}
+				defer os.RemoveAll(VolumePath)
+
+				mockMounter.EXPECT().PathExists(VolumePath).Return(true, nil)
+
+				awsDriver := nodeService{
+					mounter: mockMounter,
+				}
+
+				req := &csi.NodeGetVolumeStatsRequest{
+					VolumeId:   volumeID,
+					VolumePath: VolumePath,
+				}
+				_, err = awsDriver.NodeGetVolumeStats(context.TODO(), req)
+				if err != nil {
+					t.Fatalf("Expect no error but got: %v", err)
+				}
+			},
+		},
+		{
+			name: "fail path not exist",
+			testFunc: func(t *testing.T) {
+				mockCtl := gomock.NewController(t)
+				defer mockCtl.Finish()
+
+				mockMounter := NewMockMounter(mockCtl)
+				VolumePath := "/test"
+
+				mockMounter.EXPECT().PathExists(VolumePath).Return(false, nil)
+
+				awsDriver := nodeService{
+					mounter: mockMounter,
+				}
+
+				req := &csi.NodeGetVolumeStatsRequest{
+					VolumeId:   volumeID,
+					VolumePath: VolumePath,
+				}
+				_, err := awsDriver.NodeGetVolumeStats(context.TODO(), req)
+				expectErr(t, err, codes.NotFound)
+			},
+		},
+		{
+			name: "fail can't determine block device",
+			testFunc: func(t *testing.T) {
+				mockCtl := gomock.NewController(t)
+				defer mockCtl.Finish()
+
+				mockMounter := NewMockMounter(mockCtl)
+				VolumePath := "/test"
+
+				mockMounter.EXPECT().PathExists(VolumePath).Return(true, nil)
+
+				awsDriver := nodeService{
+					mounter: mockMounter,
+				}
+
+				req := &csi.NodeGetVolumeStatsRequest{
+					VolumeId:   volumeID,
+					VolumePath: VolumePath,
+				}
+				_, err := awsDriver.NodeGetVolumeStats(context.TODO(), req)
+				expectErr(t, err, codes.Internal)
+			},
+		},
+		{
+			name: "fail error calling existsPath",
+			testFunc: func(t *testing.T) {
+				mockCtl := gomock.NewController(t)
+				defer mockCtl.Finish()
+
+				mockMounter := NewMockMounter(mockCtl)
+				VolumePath := "/test"
+
+				mockMounter.EXPECT().PathExists(VolumePath).Return(false, errors.New("get existsPath call fail"))
+
+				awsDriver := nodeService{
+					mounter: mockMounter,
+				}
+
+				req := &csi.NodeGetVolumeStatsRequest{
+					VolumeId:   volumeID,
+					VolumePath: VolumePath,
+				}
+				_, err := awsDriver.NodeGetVolumeStats(context.TODO(), req)
+				expectErr(t, err, codes.Internal)
```

Comment:
nit: maybe it's a good idea to check the error messages too as `codes.Internal` is used for multiple cases.

Reply:
Good idea. We can improve this later.

Reply:
Will update in this PR to check the error msg.